pkg/controllers/secret/secretsync.go

```diff
@@ -0,0 +1,159 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/csi-driver-manila-operator/pkg/util"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
+)
+
+// This Controller translates Secret provided by cloud-credential-operator into
+// format required by the CSI driver.
+type Controller struct {
+	operatorClient v1helpers.OperatorClient
+	kubeClient     kubernetes.Interface
+	secretLister   corelisters.SecretLister
+	eventRecorder  events.Recorder
+}
+
+const (
+	// Name of key with clouds.yaml in Secret provided by cloud-credentials-operator.
+	cloudSecretKey = "clouds.yaml"
+	// Name of OpenStack in clouds.yaml
+	cloudName = "openstack"
+)
+
+func NewController(
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	informers v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder) factory.Controller {
+
+	// Produce secrets in the operator namespace
+	secretInformer := informers.InformersFor(util.OperatorNamespace)
+	c := &Controller{
+		operatorClient: operatorClient,
+		kubeClient:     kubeClient,
+		secretLister:   secretInformer.Core().V1().Secrets().Lister(),
+		eventRecorder:  eventRecorder,
+	}
+	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).WithInformers(
+		operatorClient.Informer(),
+		secretInformer.Core().V1().Secrets().Informer(),
+	).ToController("SecretSync", eventRecorder)
```

Comment:
There's no resync interval set with `ResyncEvery()` so this controller won't resync periodically.

Reply:
Added 20 minutes resync